sway/tree/container.c

```diff
@@ -1226,10 +1228,13 @@ void container_add_gaps(struct sway_container *c) {
 		}
 	}
 
-	c->current_gaps.top = c->y == ws->y ? ws->gaps_inner : 0;
-	c->current_gaps.right = ws->gaps_inner;
-	c->current_gaps.bottom = ws->gaps_inner;
-	c->current_gaps.left = c->x == ws->x ? ws->gaps_inner : 0;
+	int gaps_horizontal = MAX(0, ws->gaps_inner + MIN(0, c->width - ws->gaps_inner - MIN_SANE_W));
+	c->current_gaps.left = gaps_horizontal / 2;
+	c->current_gaps.right = gaps_horizontal - (gaps_horizontal / 2);
```

Comment:
I did that initially. But in `workspace.c` that doesn't work because the gaps are added with `+=` instead of set to the value. So to make everything parallel I did it like this. The performance impact is surely irrelevant but I can go back to calculating `gaps_horizontal1`, and `gaps_horizontal2` the same way in both locations and only then doing the assignments. It makes the code less compact but may be easier to understand.